src/macaron/slsa_analyzer/package_registry/pypi_registry.py

```diff
@@ -0,0 +1,198 @@
+        """
+        version: str | None = self.base_metadata.get("info", None).get("version", None)
+        return version
+
+    def get_sourcecode_url(self) -> str | None:
+        """Get the url of the source distribution.
+
+        Returns
+        -------
+            str | None: Url of the source distribution.
+        """
+        for distribution in self.base_metadata.get("urls", []):
+            if distribution.get("python_version") == "source":
+                source: str = distribution.get("url", "")
+                if source:
+                    return source
+        return None
+
+    def get_latest_release_upload_time(self) -> str | None:
+        """Get upload time of the latest release.
+
+        Returns
+        -------
+            str | None: Upload time of latest release.
+        """
+        distributions = self.base_metadata.get("urls", None)
+        return distributions[0].get("upload_time") if distributions is not None else None
+
+    def get_package_page(self) -> str | None:
+        """Implement custom API to get package main page.
+
+        Returns
+        -------
+            str | None: Package main page.
+        """
+        url = os.path.join(self.base_url, "project", self.package)
+        response = send_get_http_raw(url, headers=self.headers)
+        if response:
+            html_snippets = response.content.decode("utf-8")
+            return html_snippets
+        return None
+
+    def get_maintainer_of_package(self) -> list | None:
+        """Implement custom API to get all maintainers of the package.
+
+        Returns
+        -------
+            list | None: Maintainers.
+        """
+        package_page: str | None = self.get_package_page()
+        if package_page is None:
+            return None
+        soup = BeautifulSoup(package_page, "html.parser")
+        maintainers = soup.find_all("span", class_="sidebar-section__user-gravatar-text")
+        return list({maintainer.get_text(strip=True) for maintainer in maintainers})
+
+    def get_maintainer_profile_page(self, username: str) -> str | None:
+        """Implement custom API to get maintainer's profile page.
+
+        Parameters
+        ----------
+            username (str): Maintainer's name.
+
+        Returns
+        -------
+            str | None: Profile page.
+        """
+        url = os.path.join(self.base_url, "user", username)
+        response = send_get_http_raw(url, headers=self.headers)
+        if response:
+            html_snippets = response.content.decode("utf-8")
+            return html_snippets
+        return None
+
+    def get_maintainer_join_date(self, maintainer: str) -> datetime | None:
+        """Implement custom API to get the maintainers join date.
+
+        Parameters
+        ----------
+            maintainer (_type_): Username.
+
+        Returns
+        -------
+            datetime | None: Maintainers join date. Only recent maintainer's data available.
```

Comment:
```suggestion
        datetime | None: 
            Maintainers join date. Only recent maintainer's data available.
```

Reply:
It would be good to go through all the doc strings again when you have finalized the implementation following this guide - https://oracle.github.io/macaron/pages/developers_guide/style_guide.html